src/main/software/amazon/event/ruler/MatchType.java

```diff
@@ -14,5 +14,6 @@ public enum MatchType {
     ANYTHING_BUT,        // black list effect
     ANYTHING_BUT_PREFIX, // anything that doesn't start with this
     EQUALS_IGNORE_CASE,  // case-insensitive string match
-    WILDCARD             // string match using one or more non-consecutive '*' wildcard characters
-}
+    WILDCARD,             // string match using one or more non-consecutive '*' wildcard characters
+    NO
```

Comment:
Can you elaborate why we have `NO` here?

Reply:
Yeah, because in Parser's method #parse(MatchType type, String value) we need to pass MatchType, to prevent using [```null```](https://www.yegor256.com/2014/05/13/why-null-is-bad.html), I suggest to use [Null Object Pattern](https://www.baeldung.com/java-null-object-pattern) instead.

Reply:
> Can you elaborate why we have `NO` here?

Yeah, because in Parser's method #parse(MatchType type, String value) we need to pass MatchType, to prevent using [```null```](https://www.yegor256.com/2014/05/13/why-null-is-bad.html), I suggest to use [Null Object Pattern](https://www.baeldung.com/java-null-object-pattern) instead.